javascript/javascript1/week3/homework.js

```diff
@@ -0,0 +1,268 @@
+
+const seriesDurations = [
+	{
+		title: "House of Cards",
+		days: 2,
+		hours: 12,
+		minutes: 50,
+	},
+	{
+		title: "Ozark",
+		days: 1,
+		hours: 17,
+		minutes: 4,
+	},
+	{
+		title: "Tiger King",
+		days: 0,
+		hours: 9,
+		minutes: 19,
+	},
+];
+
+/*seriesDurations.push({
+	title: "House of Cards",
+	days: 2,
+	hours: 16,
+	minutes: 29,
+});
+*/
+
+function logOutSeriesText(seriesDurations) {
+	// write code here
+	let sumPercentage = 0;
+	for (let i = 0; i < seriesDurations.length; i++) {
+		const timeInMinutes =
+			seriesDurations[i].days * 24 * 60 +
+			seriesDurations[i].hours * 60 +
+			seriesDurations[i].minutes * 60;
+		console.log(timeInMinutes);
+		const percentage = (timeInMinutes / (80 * 525600)) * 100;
+		console.log(
+			`${seriesDurations[i].title} took ${percentage.toFixed(4)} of my life`
+		);
+		sumPercentage += percentage;
+	}
+	console.log(`In total that is ${sumPercentage.toFixed(4)} of my life`);
+}
+
+logOutSeriesText(seriesDurations);
+
+//NONON0NOYES (NOTE TAKING APP)
+
+const notes = [];
+function saveNote(content, id) {
+	if (typeof content !== "string" || content === "") {
+		throw new Error("content must be a non-empty string");
+	}
+	if (typeof id !== "number" || id === "") {
+		throw new Error("id must be a non-empty number");
+	}
+	notes.push({
+		content,
+		id,
+	});
+}
+
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+
+console.log(notes);
+
+function getNote(id) {
+	// your code here
+	if (typeof id !== "number" || id === "") {
+		throw new Error("id must be a non-empty number");
+	}
+	for (let i = 0; i < notes.length; i++) {
+		if (notes[i].id === id) {
+			return notes[i];
+		}
+	}
+}
+
+const firstNote = getNote(1);
+console.log(firstNote); // {content: 'Pick up groceries', id: 1}
+
+function logOutNotesFormatted() {
+	// your code here
+	for (let j = 0; j < notes.length; j++) {
+		console.log(
+			`The note with id: ${notes[j].id}, has the following note text: ${notes[j].content}`
+		);
+	}
+}
+
+logOutNotesFormatted(); // should log out the text below
+
+// The note with id: 1, has the following note text: Pick up groceries
+// The note with id: 2, has the following note text: Do laundry
+
+//Delete note
+
+function deleteNote(id) {
+	if (typeof id !== "number" || id === "") {
+		throw new Error("id must be a non-empty number");
+	}
+	for (let t = 0; t < notes.length; t++) {
+		if (notes[t].id === id) {
+			notes.splice(id - 1, 1);
+		}
+	}
+}
+
+deleteNote(2);
+
+console.log(notes);
+
+//CACTUSIO-INTERACTIVE (SMART PHONE USAGE APP)
+
+const activities = [];
+let limit;
+
+function addActivity(activity, duration) {
+	if (typeof activity !== "string") {
+		throw new Error("Activity must be a string");
+	}
+	if (typeof duration !== "number") {
+		throw new Error("Duration must be a number");
+	}
+
+	const today = new Date();
+	const date = today.toLocaleDateString("en-US");
+
+	const oneActivity = {
+		date,
+		activity,
+		duration,
+	};
+
+	activities.push(oneActivity);
```

Comment:
I would suggest not modifying the global array `activities` within your function. A better approach would be to just return the `oneActivity` object from the function and then, add it to the `activities` outside the function:

In general, it's not recommended to use Global variables unless you really need it (You can read more about it [here](https://www.geeksforgeeks.org/why-to-avoid-global-variables-in-javascript/)). Even if you do, try to avoid modifying it within a function block.

Let me know if you have any doubts or questions about it:)

Reply:
didn't really get this one :)
You suggest 
`return oneActivity` in the function.
and then `activities.push(oneActivity)` after the function?
Then function name doesn't make sense. It should be createActivity. Or I should just use local variable?

Reply:
Yes, I see your point here. It's fine for now then because the assignment expects you to do it in that way. I can see the same thing being done in the `saveNote` function. 

So, just take a note of it - it would be better especially when you'll work with a slightly larger codebase.